### ISSUE TITLE
Incorrect evals output when used via XGBClassifier interface

### ISSUE THREAD

**Issue by lwoloszy** [dmlc/xgboost#725]
I was getting some funky behavior when using evals via XGBClassifier. I believe it's due to line 322 in sklearn.py. When creating the evals DMatrix, missing should be set to self.missing, the same as in train_dmatrix.

**Comment by terrytangyuan** [comment]
Could you provide some reproducible example? Pull requests are more than welcomed!

**Comment by lwoloszy** [comment]
```python
    
import xgboost as xgb
import numpy as np

from sklearn.metrics import roc_auc_score
from sklearn.cross_validation import StratifiedKFold

np.random.seed(10)
X = np.random.randint(-1, 2, 100)[:, np.newaxis]
y = np.int64(np.random.rand(100) < .5)

train, valid = list(StratifiedKFold(y, 4))[1]
X_train = X[train]
X_valid = X[valid]
y_train = y[train]
y_valid = y[valid]

clf = xgb.XGBClassifier(n_estimators=1, objective='binary:logistic', silent=True)

clf.fit(X_train, y_train, eval_metric='auc',
         eval_set=[(X_valid, y_valid)])
preds = clf.predict_proba(X_valid)[:, 1]
print('The roc score reported by roc_auc_score: {}'.format(roc_auc_score(y_valid, preds)))
```

The validation_0 auc and roc_auc_score should be the same

**Comment by terrytangyuan** [comment]
@lwoloszy Thanks! I've made a fix at 7606bf8. Could you try it again?

**Comment by lwoloszy** [comment]
perfect
